src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1320,7 +1320,21 @@ internal override void SetDocumentContext(DocumentId documentId)
 
         internal bool TryGetHierarchy(ProjectId projectId, out IVsHierarchy hierarchy)
         {
-            hierarchy = this.GetHierarchy(projectId);
+            hierarchy = GetHierarchy(projectId);
+            if (hierarchy != null)
+            {
+                return true;
+            }
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            var project = _projectTracker.GetProject(projectId);
```

Comment:
What's the plan for this once we finally move TypeScript off of this shim? Can we add a comment describing what that plan would be here?

Reply:
(I understand this is a useful tactical fix which is good for now.)